lib/griddler/sendgrid/adapter.rb

```diff
@@ -27,11 +27,13 @@ def recipients(key)
       end
 
       def attachment_files
-        params.delete('attachment-info')
+        info = JSON.parse params.delete('attachment-info')
         attachment_count = params[:attachments].to_i
-
         attachment_count.times.map do |index|
-          params.delete("attachment#{index + 1}".to_sym)
+          params.delete("attachment#{index + 1}".to_sym).tap{|file|
+            # fix filename
+            file.original_filename = info["attachment#{index + 1 }"]['filename']
```

Comment:
However it might be worthwhile to guard against some part of `info["attachment#{index + 1 }"]['filename']` being `nil`, since it seems like we can't really count on sendgrid for much consistency...?